external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisUtils.scala

```diff
@@ -73,7 +73,7 @@ object KinesisUtils {
     // Setting scope to override receiver stream's scope of "receiver stream"
     ssc.withNamedScope("kinesis stream") {
       new KinesisInputDStream[T](ssc, streamName, endpointUrl, validateRegion(regionName),
-        initialPositionInStream, kinesisAppName, checkpointInterval, storageLevel,
+        initialPositionInStream, null, kinesisAppName, checkpointInterval, storageLevel,
```

Comment:
Passing a ```null``` for an optional value is very un-idiomatic for Scala code. As noted in my previous comment, we should find a way to pass a single config option that represents all possible InitialPositionInStream configurations or at the very least make this argument an ```Option[Date]``` with a default value of ```None```.